WordPressVIPMinimum/Sniffs/JS/HTMLExecutingFunctionsSniff.php

```diff
@@ -58,29 +71,56 @@ public function register() {
 	 */
 	public function process_token( $stackPtr ) {
 
-		if ( false === in_array( $this->tokens[ $stackPtr ]['content'], $this->HTMLExecutingFunctions, true ) ) {
+		if ( ! isset( $this->HTMLExecutingFunctions[ $this->tokens[ $stackPtr ]['content'] ] ) ) {
 			// Looking for specific functions only.
 			return;
 		}
 
-		$nextToken = $this->phpcsFile->findNext( Tokens::$emptyTokens, $stackPtr + 1, null, true, null, true );
+		if ( 'content' === $this->HTMLExecutingFunctions[ $this->tokens[ $stackPtr ]['content'] ] ) {
+			$nextToken = $this->phpcsFile->findNext( Tokens::$emptyTokens, $stackPtr + 1, null, true, null, true );
 
-		if ( T_OPEN_PARENTHESIS !== $this->tokens[ $nextToken ]['code'] ) {
-			// Not a function.
-			return;
-		}
+			if ( T_OPEN_PARENTHESIS !== $this->tokens[ $nextToken ]['code'] ) {
+				// Not a function.
+				return;
+			}
+
+			$parenthesis_closer = $this->tokens[ $nextToken ]['parenthesis_closer'];
 
-		$parenthesis_closer = $this->tokens[ $nextToken ]['parenthesis_closer'];
+			while ( $nextToken < $parenthesis_closer ) {
+				$nextToken = $this->phpcsFile->findNext( Tokens::$emptyTokens, $nextToken + 1, null, true, null, true );
+				if ( T_STRING === $this->tokens[ $nextToken ]['code'] ) { // Contains a variable.
+					$message = 'Any HTML passed to `%s` gets executed. Make sure it\'s properly escaped.';
+					$data    = [ $this->tokens[ $stackPtr ]['content'] ];
+					$this->phpcsFile->addWarning( $message, $stackPtr, $this->tokens[ $stackPtr ]['content'], $data );
 
-		while ( $nextToken < $parenthesis_closer ) {
-			$nextToken = $this->phpcsFile->findNext( Tokens::$emptyTokens, $nextToken + 1, null, true, null, true );
-			if ( T_STRING === $this->tokens[ $nextToken ]['code'] ) {
-				$message = 'Any HTML passed to `%s` gets executed. Make sure it\'s properly escaped.';
-				$data    = [ $this->tokens[ $stackPtr ]['content'] ];
-				$this->phpcsFile->addWarning( $message, $stackPtr, $this->tokens[ $stackPtr ]['content'], $data );
+					return;
+				}
+			}
+		} elseif ( 'target' === $this->HTMLExecutingFunctions[ $this->tokens[ $stackPtr ]['content'] ] ) {
+			$prevToken = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $stackPtr - 1, null, true, null, true );
 
+			if ( T_OBJECT_OPERATOR !== $this->tokens[ $prevToken ]['code'] ) {
 				return;
 			}
+
+			$prevPrevToken = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $prevToken - 1, null, true, null, true );
+
+			if ( T_CLOSE_PARENTHESIS !== $this->tokens[ $prevPrevToken ]['code'] ) {
```

Comment:
I believe that we should also check whether the keyword we have found is a function call. As, for instance, the code, as it is now, would flag following code:

```
foo( variable ).appendTo;
```